src/Microsoft.Azure.WebJobs.Host/FunctionTimeoutException.cs

```diff
@@ -33,17 +34,23 @@ protected FunctionTimeoutException(SerializationInfo info, StreamingContext cont
         }
 
         /// <inheritdoc/>
-        public FunctionTimeoutException(string message, Guid instanceId, string methodName, TimeSpan timeout, Exception innerException)
+        public FunctionTimeoutException(string message, Guid instanceId, string methodName, TimeSpan timeout, Task functionTask, Exception innerException)
```

Comment:
We'll use it in Script. We don't use it anywhere here (but we could). In Script, we have a custom IWebJobsExceptionHandler that will waita few seconds to see if the Task exits gracefully. If it does, we can continue on without restarting the site. See here for usage: https://github.com/Azure/azure-webjobs-sdk-script/pull/708